peer.go

```diff
@@ -76,9 +76,20 @@ func newPeerList(root *RootPeerList) *PeerList {
 	}
 }
 
-// SetStrategy sets customized peer selection stratedgy.
+// SetStrategy sets customized peer selection strategy.
 func (l *PeerList) SetStrategy(sc ScoreCalculator) {
+	l.Lock()
+	defer l.Unlock()
 	l.scoreCalculator = sc
```

Comment:
You mean `.exists()` should also return the score calculator?

Well that help with the common function, since that function assumes that an outer lock is held, since the lock is held for a *long* time in `SetStrategy()`.